qcodes/instrument_drivers/ZI/ZIUHFLI.py

```diff
@@ -718,8 +718,9 @@ def __init__(self, name: str, device_ID: str, **kwargs) -> None:
                                                oscs-1, 0, 'oscselet'),
                                set_cmd=partial(self._setter, 'demods',
                                                oscs-1, 0, 'oscselect'),
-                               val_mapping={i+1: i for i in range(8)},
-                               vals=vals.Ints(1, 8))
+                               val_mapping={i + 1: i for i in
+                                            range(number_of_oscillators)},
+                               vals=vals.Ints(1, number_of_oscillators))
```

Comment:
This is actually redundant, the `val_mapping` creates a default validator which is just an `Enum` validator of the dict keys.

Reply:
I removed the vals